sklearn/cluster/tests/test_spectral.py

```diff
@@ -412,3 +413,33 @@ def new_svd(*args, **kwargs):
 
     with pytest.raises(LinAlgError, match="SVD did not converge"):
         discretize(vectors)
+
+
+# WIP
+def test_spectral_eigen_tol_auto():
+    """Test that `eigen_tol="auto"` is resolved correctly"""
```

Comment:
I'm not sure what the best way to test that `spectral_clustering` internally passes the correct tolerance values to its solvers based on `eigen_tol="auto"` -- it's not enough to compare against the output if we manually set `eigen_tol` to the expected auto-resolved value, since it can easily reproduce similar results. Open too ideas

Reply:
For this specific case, I would [monkeypatch](https://docs.pytest.org/en/6.2.x/monkeypatch.html) either "arpack" or "lobpcg" with a [Mock](https://docs.python.org/3/library/unittest.mock.html). This will tests that "auto" is mapping the to correct value.

Here is a simple example of mocking and asserting the input value:

```python
from unittest.mock import Mock

def add_one(x):
    return x + 1

mocked_add_one = Mock(side_effect=add_one)
mocked_add_one(3)

mocked_add_one.assert_called_with(3)
```